api/types/accesslist/accesslist.go

```diff
@@ -251,6 +251,17 @@ type Status struct {
 	OwnerOf []string `json:"owner_of" yaml:"owner_of"`
 	// MemberOf is a list of Access List UUIDs where this access list is an explicit member.
 	MemberOf []string `json:"member_of" yaml:"member_of"`
+
+	// SelfAccess describes the current user's ownership and membership in the access list.
+	SelfAccess *SelfAccess `json:"-" yaml:"-"`
+}
+
+// SelfAccess describes the current user's ownership and membership in the access list.
+type SelfAccess struct {
+	// OwnershipType represents the current user's ownership type (explicit, inherited, or none) in the access list.
+	OwnershipType uint32 `json:"ownership_type" yaml:"ownership_type"`
+	// MembershipType represents the current user's membership type (explicit, inherited, or none) in the access list.
+	MembershipType uint32 `json:"membership_type" yaml:"membership_type"`
```

Comment:
Could you create proper enum types for these fields  ?

Reply:
Is it possible to re-use the enum from `accesslists` package here?